packages/js/uri-resolvers/src/utils/getUriResolutionPath.ts

```diff
@@ -0,0 +1,42 @@
+import { IUriResolutionStep } from "@polywrap/core-js";
+
+export const getUriResolutionPath = (
+  history: IUriResolutionStep<unknown>[]
+): IUriResolutionStep<unknown>[] => {
+  // Get all non-empty items from the resolution history
+  return (
+    history
+      .filter((x) => {
+        // If a failure happened
+        if (!x.result.ok) {
+          return true;
+        }
+
+        const uriPackageOrWrapper = x.result.value;
+
+        if (uriPackageOrWrapper.type === "uri") {
+          // If a redirect has occured
+          return uriPackageOrWrapper.uri.uri !== x.sourceUri.uri;
+        } else if (uriPackageOrWrapper.type === "package") {
```

Comment:
nit: I am noticing that we are using a lot the strings `package` and `wrapper` - Would it make sense to create an `Enum` somewhere and just use the values of the Enum? Making the code easier to maintain

Reply:
If it was another language then I'd say yes, definitely. But I find that in TypeScript string unions (not plain strings) are preferable since they don't require another enum type, don't require importing that type, are more readable and (unlike in other languages) are type safe

Reply:
Yah I agree, enums are a bit tricky in TypeScript. Here's an article that might cover some of the erroneous cases (didn't read myself): https://thoughtbot.com/blog/the-trouble-with-typescript-enums